index.d.ts

```diff
@@ -211,11 +227,144 @@ export interface DailyRoomInfo {
   dialInPIN?: string;
 }
 
-export interface DailyEventObject {
-  action: string;
-  [payloadProp: string]: any;
+export interface DailyEventObjectNoPayload {
+  action: Extract<
+    DailyEvent,
+    | 'loading'
+    | 'loaded'
+    | 'joining-meeting'
+    | 'left-meeting'
+    | 'recording-started'
+    | 'recording-stopped'
+    | 'recording-stats'
+    | 'recording-error'
+    | 'recording-upload-completed'
+    | 'fullscreen'
+    | 'exited-fullscreen'
+    | 'live-streaming-started'
+    | 'live-streaming-stopped'
+    | 'track-started'
+    | 'track-stopped'
+  >;
+}
+
+export interface DailyEventErrorObject {
+  action: Extract<
+    DailyEvent,
+    'load-attempt-failed' | 'live-streaming-error' | 'error'
+  >;
+  errorMsg: string;
+}
+
+export interface DailyEventObjectParticipants {
+  action: Extract<DailyEvent, 'joined-meeting'>;
+  participants: DailyParticipantsObject;
+}
+
+export interface DailyEventObjectParticipant {
+  action: Extract<
+    DailyEvent,
+    'participant-joined' | 'participant-updated' | 'participant-left'
+  >;
+  participant: DailyParticipant;
+}
+
+export interface DailyEventObjectMouseEvent {
+  action: Extract<
+    DailyEvent,
+    'click' | 'mousedown' | 'mouseup' | 'mouseover' | 'mousemove'
+  >;
+  event: {
```

Comment:
Updated this `event` property to not reference `MouseEvent` but rather the synthetic event that we actually provide, which is mostly comprised of properties pulled off of `MouseEvent`. Same goes for `DailyEventObjectTouchEvent`.